winui/DataGrid/Export-To-Excel.md

```diff
@@ -0,0 +1,653 @@
+    {
+        e.Style.Color = Color.FromArgb(252, 159, 161);
+        e.Handled = true;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-cell-style-customization.png" alt="WinUI DataGrid displays Customized Cell Style based on CellType in Exported Excel" width="100%" Height="Auto"/>
+
+## Cell customization in Excel while exporting
+
+You can customize the cells by setting `CellsExportHandler` in `DataGridExcelExportOptions`.
+
+### Customize cell value while exporting
+
+You can customize the call values while exporting to excel by using `CellsExportHandler` in `DataGridExcelExportOptions`.
+
+{% tabs %}
+{% highlight c# %}
+var options = new DataGridExcelExportOptions();
+options.CellsExportHandler = CellsExportHandler;
+var excelEngine = dataGrid.ExportToExcel(dataGrid.View, options);
+var workBook = excelEngine.Excel.Workbooks[0];
+MemoryStream stream = new MemoryStream();
+workBook.SaveAs(stream);
+Save(stream, "Sample");
+
+private static void CellsExportHandler(object sender, DataGridCellExcelExportOptions e)
+{
+    // Based on the column mapping name and the cell type, we can change the cell 
+    //values while exporting to excel.
+    if (e.CellType == ExportCellType.RecordCell && e.ColumnName == "IsShipped")
+    {
+
+        //if the cell value is True, "Y" will be displayed else "N" will be displayed.
+
+        if (e.CellValue.Equals(true))
+            e.Range.Cells[0].Value = "Y";
+
+        else
+            e.Range.Cells[0].Value = "N";
+        e.Handled = true;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-cell-value-customization.png" alt="Customizing Cell Values while Exporting to Excel in WinUI DataGrid" width="100%" Height="Auto"/>
+
+Here, cell values are changed for `Is Shipped` column based on custom condition.
+
+### Changing row style in excel based on data
+
+You can customize the rows based on the record values by using `CellsExportHandler`.
+
+{% tabs %}
+{% highlight c# %}
+var options = new DataGridExcelExportOptions();
+options.CellsExportHandler = CellsExportHandler;
+var excelEngine = dataGrid.ExportToExcel(dataGrid.View, options);
+var workBook = excelEngine.Excel.Workbooks[0];
+MemoryStream stream = new MemoryStream();
+workBook.SaveAs(stream);
+Save(stream, "Sample");
+
+private static void CellsExportHandler(object sender, DataGridCellExcelExportOptions e)
+{
+    var record = e.NodeEntry;
+
+    if (record != null && (record as OrderInfo).Country == "Mexico")
+    {
+        e.Range.CellStyle.ColorIndex = ExcelKnownColors.Green;
+        e.Range.CellStyle.Font.Color = ExcelKnownColors.White;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img src="Export-to-Excel_images/winui-datagrid-row-style-customization.png" alt="Customizing Row Style based on Data while Exporting to Excel in WinUI DataGrid" width="100%" Height="Auto"/>
+
+Here, records having the `Country` name as `Mexico` are customized.
```

Comment:
Here, the records having the `Country` name as `Mexico` are customized.

Reply:
Content modified